operator-common/src/main/java/io/strimzi/operator/common/auth/PemAuthIdentity.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.auth;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.common.Util;
+
+import java.io.ByteArrayInputStream;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Objects;
+
+/**
+ * Represents the identity used during TLS client authentication.
+ * This consists of an X509 end-entity certificate, corresponding private key, and a (possibly empty) chain of X509 intermediate CA certificates, all in PEM format.
+ */
+public class PemAuthIdentity {
+
+    private final byte[] privateKeyAsPemBytes;
+    private final byte[] certificateChainAsPemBytes;
+    private final String secretCertName;
+    private final String secretName;
+    private final String secretNamespace;
+
+    /**
+     * @param secret Kubernetes Secret containing the Cluster Operator public and private key
```

Comment:
MAybe you could add some description if you add some Javadoc?

Reply:
I've updated the Javadocs so they all have descriptions